pallets/valve/src/lib.rs

```diff
@@ -86,26 +104,68 @@ pub mod pallet {
 
 			ValveClosed::<T>::put(true);
 			<Pallet<T>>::deposit_event(Event::ValveClosed);
-			Ok(().into())
+			Ok(())
+		}
+
+		/// Tap the valve.
+		///
+		/// If valve is closed you cannot tap a pallet.
+		/// You cannot tap this paller, as then you could never untap it.
+		#[pallet::weight(T::DbWeight::get().read + 2 * T::DbWeight::get().write)]
+		pub fn tighten_valve(origin: OriginFor<T>, pallet_name: Vec<u8>) -> DispatchResult {
+			ensure_root(origin)?;
+
+			// Ensure the valve isn't already off.
+			// If the valve is already off there is no need to tap individual pallets.
+			ensure!(!ValveClosed::<T>::get(), Error::<T>::ValveAlreadyClosed);
+
+			let pallet_name_string =
+				sp_std::str::from_utf8(&pallet_name).map_err(|_| Error::<T>::InvalidCharacter)?;
+
+			// Not allowed to pause calls of this pallet as then you could never unpause them.
+			ensure!(
+				pallet_name_string != <Self as PalletInfoAccess>::name(),
+				Error::<T>::CannotPause
+			);
+
+			TappedPallets::<T>::mutate_exists(pallet_name.clone(), |maybe_tapped| {
```

Comment:
curious why we keep a tapped pallets vector. shouldn't we have a separate pallets vector already that we keep track of, where each pallet knows if it's tapped or not?

Reply:
a Vec<u8> is a vector of btyes. it's another way to have a string